odbc/src/api/data_tests.rs

```diff
@@ -158,2196 +165,2239 @@ lazy_static! {
                     Schema::Atomic(Atomic::Scalar(BsonTypeName::Long)),
                     Nullability::NO_NULLS,
+                    TypeMode::Standard
                 ),
                 MongoColMetadata::new(
                     "",
                     "test".to_string(),
                     "unit_str".to_string(),
                     Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
                     Nullability::NO_NULLS,
+                    TypeMode::Standard
                 ),
                 MongoColMetadata::new(
                     "",
                     "test".to_string(),
                     "guid".to_string(),
                     Schema::Atomic(Atomic::Scalar(BsonTypeName::BinData)),
                     Nullability::NO_NULLS,
+                    TypeMode::Standard
                 ),
             ],
         );
-}
-
-mod unit {
-    use std::cell::RefCell;
-
-    use cstr::WideChar;
-
-    use super::*;
-    // test unallocated_statement tests SQLFetch when the mongo_statement inside
-    // of the statement handle has not been allocated (before an execute or tables function
-    // has been called).
-    #[test]
-    fn unallocated_statement_sql_fetch() {
-        let env = Box::into_raw(Box::new(MongoHandle::Env(Env::with_state(
-            EnvState::ConnectionAllocated,
-            RefCell::new(None),
-        ))));
-        let conn = Box::into_raw(Box::new(MongoHandle::Connection(Connection::with_state(
-            env as *mut _,
-            ConnectionState::Connected,
-        ))));
-        let stmt = Statement::with_state(conn as *mut _, StatementState::Allocated);
-        let stmt_handle: *mut _ = &mut MongoHandle::Statement(stmt);
-
-        unsafe {
-            assert_eq!(SqlReturn::ERROR, SQLFetch(stmt_handle as *mut _,));
-            assert_eq!(
-                format!("[MongoDB][API] No ResultSet"),
-                format!(
-                    "{}",
-                    (*stmt_handle)
-                        .as_statement()
-                        .unwrap()
-                        .errors
-                        .read()
-                        .unwrap()[0]
-                ),
-            );
-            let _ = Box::from_raw(conn as *mut WChar);
-            let _ = Box::from_raw(env as *mut WChar);
-        }
-    }
 
-    #[test]
-    fn sql_fetch_and_more_results_basic_functionality() {
-        let env = Box::into_raw(Box::new(MongoHandle::Env(Env::with_state(
-            EnvState::ConnectionAllocated,
-            RefCell::new(None),
-        ))));
-        let conn = Box::into_raw(Box::new(MongoHandle::Connection(Connection::with_state(
-            env as *mut _,
-            ConnectionState::Connected,
-        ))));
-        let stmt = Statement::with_state(conn as *mut _, StatementState::Allocated);
-        *stmt.mongo_statement.write().unwrap() = Some(Box::new(MongoQuery::new(
+    static ref SIMPLE_BSON_TYPE_MQ: MongoQuery = MongoQuery::new(
```

Comment:
I had to keep `SIMPLE_BSON_TYPE_MQ` for testing. The only difference between the two is the `TypeMode`.